### PR TITLE
feat(promote to failure): Add the ability to track failures across multiple jobs

### PR DESCRIPTION
**Pull request by jchauncey**
* promote to failure when not job is not being run in circleci

### CODE REVIEW

File: test/acs-engine-test/promoteToFailure/promoteToFailure.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
+package promoteToFailure
```
Comment (jackfrancis) [inline test/acs-engine-test/promoteToFailure/promoteToFailure.go]: We should change this to adhere to golang idioms (specifically, `anUpperCaseName` is not recommended). See:

https://blog.golang.org/package-names

How about `ptf`?

File: test/acs-engine-test/promoteToFailure/promoteToFailure.go [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	azureStorage "github.com/Azure/azure-sdk-for-go/storage"
```
Comment (jackfrancis) [inline test/acs-engine-test/promoteToFailure/promoteToFailure.go]: ditto with the alias `azureStorage`, let's get rid of the `upperCaseName` thing here

File: test/acs-engine-test/promoteToFailure/promoteToFailure.go (hunk 1 above)
Comment (jchauncey) [reply test/acs-engine-test/promoteToFailure/promoteToFailure.go]: im trying to keep this as close to the work shruti did as possible just so its easy to compare notes when things arent working.

Comment (jchauncey) [reply test/acs-engine-test/promoteToFailure/promoteToFailure.go]: alright just changed this to be promote

File: test/acs-engine-test/promoteToFailure/promoteToFailure.go (hunk 2 above)
Comment (jchauncey) [reply test/acs-engine-test/promoteToFailure/promoteToFailure.go]: fixed

**Comment by jackfrancis** [comment]
@jchauncey fyi

```
test/e2e.sh: line 31: SA_NAME: unbound variable
Makefile:90: recipe for target 'test-e2e' failed
```

**Comment by jchauncey** [comment]
@jackfrancis yeah i had to remove the `set -u` from e2e.sh

**Pull request merged**